File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        for(String p : participant){
            // 이름을 key 값으로 생성, 동명이인 있을 경우 value 1 증가
            map.put(p, map.getOrDefault(p, 0)+1);
        }
        
        for (String c : completion) {
            map.put(c, map.get(c) - 1); // 완주했을 경우 -1
        }
        
        // 0이 아닐 경우(완주하지 못했을 경우) key 값을 뽑아서 출력
        for (String key : map.keySet()) {
        if (map.get(key) != 0) {
            answer = key;
            break;
        }
    }
        
        return answer;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
import collections

def solution(participant, completion):
    participant.sort() # 배열 정렬
    completion.sort()
    
    # Counter()는 Dic과 같이 hash형 자료들의 값의 개수를 셀 때 사용
    answer = collections.Counter(participant) - collections.Counter(completion)
    
    # 완주하지 못한 선수는 1명으로 정해져있기 때문에 리스트화 후 0번째 리턴
    return list(answer)[0]